docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+                    }
+                ]
+            },
+            {
+                // The invitation token doesn't correspond to any existing invitation
+                "status": "invitation_not_found"
+            },
+            {
+                // The invitation has already been completed
+                "status": "invitation_completed"
+            },
+            {
+                // The invitation has been cancelled
+                "status": "invitation_cancelled"
+            },
+            {
+                // The author is not part of the allowed greeters for this invitation
+                "status": "author_not_allowed"
+            },
+            {
+                // The attempt id doesn't correspond to any existing attempt
+                "status": "attempt_not_found"
+            },
+            {
+                // The author did not join the attempt
+                // This should not happen, since joining is required to get the attempt ID
+                "status": "attempt_not_joined"
+            },
+            {
+                // The attempt has been already cancelled
+                "status": "attempt_already_cancelled",
+                "fields": [
+                    {
+                        "name": "origin",
+                        "type": "GreeterOrClaimer"
+                    },
+                    {
+                        "name": "timestamp",
+                        "type": "DateTime"
+                    },
+                    {
+                        "name": "reason",
+                        "type": "CancelledAttemptReason"
+                    }
+                ]
+            }
+        ]
+    }
+]
+```
+
+
+#### `invite_claimer_cancel_attempt` command
+
+This is the new invited command introduced for the claimer to cancel an attempt with a provided reason.
+
+In order for the claimer to cancel an attempt, a couple of checks have to be performed:
+- the greeter exists
+- the greeter is allowed (i.e. it is part of the greeters)
+- the attempt exists
+- the attempt is active (i.e. joined and not cancelled)
+
+> [!NOTE]
+> The state of the invitation doesn't have to be checked since it is already checked by the `/invited` HTTP route.
+
+Otherwise this command has the same logic as `invite_greeter_cancel_attempt`.
+
+Pseudo-code:
+
+```python
+def invite_claimer_cancel_attempt(
+    token: InvitationToken,
+    greeter: GreeterID,
+    attempt: AttemptID,
+    reason: CancelledAttemptReason,
+) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  GreeterNotFound | GreeterNotAllowed | OK:
+    # Perform checks
+    [...]
+    # Call `claimer_cancel_attempt(reason)`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_claimer_cancel_attempt",
+            "fields": [
+                {
```

Comment:
Same as above: token+greeter+attemptID makes rooms for inconsistency, we should only have a `attempt` field instead

Reply:
ccf6d568d950206e8b632bc2469462e7910f790c